pkg/leaderelection/leaderelection.go

```diff
@@ -40,17 +40,19 @@ var (
 
 // Opts contains the options to configure the leader election mechanism.
 type Opts struct {
-	Enabled         bool
-	PodName         string
-	TenantNamespace string
-	LeaseDuration   time.Duration
-	RenewDeadline   time.Duration
-	RetryPeriod     time.Duration
+	Enabled           bool
+	PodName           string
+	Namespace         string
+	LeaderElectorName string
+	LeaseDuration     time.Duration
+	RenewDeadline     time.Duration
+	RetryPeriod       time.Duration
+	InitCallback      func()
+	StopCallback      func()
 }
 
 // InitAndRun initializes and runs the leader election mechanism.
-func InitAndRun(ctx context.Context, opts Opts, rc *rest.Config,
-	eb record.EventBroadcaster, initCallback func()) error {
+func InitAndRun(ctx context.Context, opts *Opts, rc *rest.Config, eb record.EventBroadcaster) error {
```

Comment:
Can you create a function that return a leaderelector and then run it inside the main instead of using a go function? You can use just one context in this way